### PR TITLE
[MLv2] Implement `automatic-insights` drill

### PR DESCRIPTION
**Pull request by bshepherdson**
[MLv2] Implement `automatic-insights` drill

Automatic insights drills have some unusual conditions.
This adds `metabase.lib.metadata/editable?` and checks it before
returning any drills.

On settings:
In the app, `MetabaseSettings` is a global singleton and the settings
are sometimes updated in place. In the JS testing environment several
mock settings instances can exist at once, and the global singleton does
not necessarily have the values we want for any given test.

This PR makes `metabase.lib.metadata/setting` check for `describe` and
`it` globals to see if this is the testing environment, and to trust the
metadata's `settings` in that case.

Implements the BE side of #33558.

### CODE REVIEW

**Comment by bshepherdson** [comment]
Current dependencies on/for this PR:
* master
  * **PR #36443** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36443?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/36443?utm_source=stack-comment).

File: test/metabase/lib/drill_thru/automatic_insights_test.cljc [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+          context       (merge agg-dim
+                               {:row        (cons agg-dim breakout-dims)
+                                :dimensions breakout-dims})]
+      (is (=? {:lib/type :mbql/query
+               :stages [{:filters     (exp-filters-fn agg-dim breakout-dims)
+                         :aggregation (symbol "nil #_\"key is not present.\"")
+                         :breakout    (symbol "nil #_\"key is not present.\"")
+                         :fields      (symbol "nil #_\"key is not present.\"")}]}
+              (->> (lib.drill-thru/available-drill-thrus query context)
+                   (m/find-first #(= (:type %) :drill-thru/underlying-records))
```
Comment (camsaul) [inline test/metabase/lib/drill_thru/automatic_insights_test.cljc:97]: All these tests look like they were copypasted from the underlying records test namespace, they're even testing for underlying records drills and not automatic-insights

**Review by camsaul — APPROVED** [review]
Approving just so we can get this in and keep the train rolling, but your tests look busted, it would be good to actually test automatic insights drills rather than just testing underlying records a second time

File: test/metabase/lib/drill_thru/automatic_insights_test.cljc (hunk 1 above)
Comment (bshepherdson) [reply test/metabase/lib/drill_thru/automatic_insights_test.cljc]: They were copy-pasted and I didn't finish porting the file. It's done now.

**Pull request merged**